homeassistant/components/upnp/device.py

```diff
@@ -134,69 +135,45 @@ def __str__(self) -> str:
         """Get string representation."""
         return f"IGD Device: {self.name}/{self.udn}::{self.device_type}"
 
-    async def async_get_traffic_data(self) -> Mapping[str, Any]:
-        """
-        Get all traffic data in one go.
-
-        Traffic data consists of:
-        - total bytes sent
-        - total bytes received
-        - total packets sent
-        - total packats received
-
-        Data is timestamped.
-        """
-        _LOGGER.debug("Getting traffic statistics from device: %s", self)
-
-        values = await asyncio.gather(
-            self._igd_device.async_get_total_bytes_received(),
-            self._igd_device.async_get_total_bytes_sent(),
-            self._igd_device.async_get_total_packets_received(),
-            self._igd_device.async_get_total_packets_sent(),
-        )
-
-        return {
-            TIMESTAMP: utcnow(),
-            BYTES_RECEIVED: values[0],
-            BYTES_SENT: values[1],
-            PACKETS_RECEIVED: values[2],
-            PACKETS_SENT: values[3],
-        }
-
-    async def async_get_status(self) -> Mapping[str, Any]:
-        """Get connection status, uptime, and external IP."""
-        _LOGGER.debug("Getting status for device: %s", self)
-
-        values = await asyncio.gather(
-            self._igd_device.async_get_status_info(),
-            self._igd_device.async_get_external_ip_address(),
-            return_exceptions=True,
-        )
-        status_info: StatusInfo | None = None
-        router_ip: str | None = None
-
-        for idx, value in enumerate(values):
-            if isinstance(value, UpnpError):
-                # Not all routers support some of these items although based
-                # on defined standard they should.
-                _LOGGER.debug(
-                    "Exception occurred while trying to get status %s for device %s: %s",
-                    "status" if idx == 1 else "external IP address",
-                    self,
-                    str(value),
-                )
-                continue
-
-            if isinstance(value, Exception):
-                raise value
-
-            if isinstance(value, StatusInfo):
-                status_info = value
-            elif isinstance(value, str):
-                router_ip = value
+    async def async_get_data(self) -> Mapping[str, Any]:
+        """Get all data from device."""
+        _LOGGER.debug("Getting data for device: %s", self)
+        igd_state = await self._igd_device.async_get_traffic_and_status_data()
 
         return {
-            WAN_STATUS: status_info[0] if status_info is not None else None,
-            ROUTER_UPTIME: status_info[2] if status_info is not None else None,
-            ROUTER_IP: router_ip,
+            TIMESTAMP: igd_state.timestamp,
+            BYTES_RECEIVED: igd_state.bytes_received
+            if igd_state.bytes_received is not None
```

Comment:
Please avoid ternary operator expressions that span more than one line. Use a regular if/else block for longer expressions.